### ISSUE TITLE
Add missing azurerm_mariadb_firewall_rule

### ISSUE THREAD

**Issue by armbraggins** [terraform-providers/terraform-provider-azurerm#3608]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add a new azurerm_mariadb_firewall_rule resource, following the model of azurerm_mysql_firewall_rule

### New or Affected Resource(s)

* azurerm_mariadb_firewall_rule

### Potential Terraform Configuration

```hcl
resource "azurerm_mariadb_firewall_rule" "mariadb_firewall_azure" {
   name                = "azure"
   resource_group_name = var.resource_group_name
   server_name         = "${azurerm_mariadb_server.mariadb_server.name}"
   start_ip_address    = "0.0.0.0"
   end_ip_address      = "0.0.0.0"
}
```

### References

* https://github.com/Azure/azure-sdk-for-go/blob/master/services/mariadb/mgmt/2018-06-01/mariadb/firewallrules.go
* https://docs.microsoft.com/en-us/cli/azure/mariadb/server/firewall-rule?view=azure-cli-latest

**Comment by armbraggins** [comment]
(I am interested in fixing this myself, but haven't used Go before, and am new to Terraform.)

**Comment by armbraggins** [comment]
Also new to making github pull requests. I have local changes that seem to work. What's the recommended workflow? Fork and create a pull request there? Ask for permission to create a branch here?

**Comment by armbraggins** [comment]
https://github.com/terraform-providers/terraform-provider-azurerm/pull/3636

**Comment by mbfrahry** [comment]
Hello again @armbraggins, it looks like this issue has been resolved through #3720 so I'll be closing this down. Thanks again for starting this up
